### PR TITLE
SaveCroppedObjects, redux

### LINKED ISSUE #2255: Create a new module: ExportImagesPerObject
This would save an individual image for each object that has been identified in an upstream module. 

This is useful for training deep learning models on individual cell images that have been segmented using CellProfiler. It can also generate the image data needed for data exploration tools designed for imaging flow cytometry, where each cell is imaged separately (though we do not yet know of any requests for this).

INPUT: 
— objects (with XY locations)
— corresponding image(s) to save (should handle RGB or grayscale, masked or not - i.e., any masking or color merging/splitting should happen upstream of this module)

SETTINGS:
— specify the input, as detailed above
— needs an “add another image” option (for multiple grayscale images, for example)
— fixed size tiles (then, specify the pixel X, Y size) or per-object size (each tile fits the object with no blank rows/columns around it) or full size image (the entire size of the original image, with the one little object in the middle - this only makes sense if the input is masked or else it saves the entire original image for every object)
— Rotate major axis of object to the XY axes of the image (two options: limit to 90deg only, vs. any angle). Default = no rotation

DISPLAY:
- nothing really needs to be displayed, but maybe could show “Number of individual images saved to disk: NNN”.  

OUTPUT: files saved to disk, plus maybe we need to save file names to the per-object measurements

### PR DESCRIPTION
**Pull request by bethac07**
Resolves #2255 

Taking the best of @SariNusier 's PR #2831  and putting it on top of the current expanded functionalities of SaveCroppedObjects.

### CODE REVIEW

File: cellprofiler/modules/savecroppedobjects.py [hunk 1]
```diff
@@ -25,27 +25,54 @@
...
+            ],
+            doc="""
+            Choose the way you want the per-object crops to be exported.
+            <p>The choices are:<br>
+            <ul><li><i>{SAVE_PER_OBJECT}</i>: Save a per-object crop from the original image based on the object's
+            bounding box.</li>
+            <li><i>{SAVE_MASK}</i>: Export a per-object mask.</li>
+            </ul></p>
+            """.format(**{
+                "SAVE_PER_OBJECT": SAVE_PER_OBJECT,
```
Comment (AetherUnbound) [inline cellprofiler/modules/savecroppedobjects.py]: I know this is how other modules do it, but for IDE ease do you mind replacing this with the following?
```
.format(
    SAVE_PER_OBJECT=SAVE_PER_OBJECT,
    SAVE_MASK=SAVE_MASK
)
```

File: cellprofiler/modules/savecroppedobjects.py [hunk 2]
```diff
@@ -86,7 +113,15 @@ def run(self, workspace):
...
         for label in unique_labels:
-            mask = labels == label
+            if self.export_option == SAVE_MASK:
+                mask = labels == label
+
+            elif self.export_option == SAVE_PER_OBJECT:
+                mask_in = labels == label
+                images = workspace.image_set
+                x = images.get_image(self.image_name.value)
+                properties=skimage.measure.regionprops(mask_in.astype(int),intensity_image=x.pixel_data)
```
Comment (AetherUnbound) [inline cellprofiler/modules/savecroppedobjects.py]: There should be spaces around the assignment and after the comma to follow pep8

File: cellprofiler/modules/savecroppedobjects.py [hunk 3]
```diff
@@ -87,7 +87,10 @@ def create_settings(self):
             ],
             value=O_TIFF,
             doc="""\
-**%(O_PNG)** files do not support 3D. **%(O_TIFF)** files use zlib compression level 6."""
+**{O_PNG}** files do not support 3D. **{O_TIFF}** files use zlib compression level 6.""".format(
```
Comment (AetherUnbound) [inline cellprofiler/modules/savecroppedobjects.py]: Nice catch!

Comment (bethac07) [reply cellprofiler/modules/savecroppedobjects.py]: Wouldn't have caught it (and other documentation problems too- some of the added documentation was in html, not rst) if you hadn't asked me to change the formatting thing, so good on you!

**Pull request merged**